src/ESports.ts

```diff
@@ -0,0 +1,172 @@
+import fetch from "node-fetch";
+import Discord = require("discord.js");
+import { SharedSettings } from "./SharedSettings";
+import * as CheerioAPI from "cheerio";
+import * as momentjs from "moment";
+import Botty from "./Botty";
+
+interface ESportsAPIReturnData {
+    resultsHtml: string;
+    fixturesHtml: string;
+    resultsMonths: string;
+    fixturesMonths: string;
+}
+
+interface ESportsLeagueSchedule {
+    league: string;
+    time: string;
+    teamA: string;
+    teamB: string;
+}
+
+export default class ESportsAPI {
+    private bot: Discord.Client;
+    private settings: SharedSettings;
+
+    private schedule: Map<string, Map<string, ESportsLeagueSchedule[]>> = new Map();
+
+    constructor(bot: Discord.Client, settings: SharedSettings) {
+        this.bot = bot;
+        this.settings = settings;
+
+        bot.on("ready", async () => {
+            await this.loadData();
+            this.postInfo();
+        });
+    }
+
+    public onCheckNext(message: Discord.Message, isAdmin: boolean, command: string, args: string[]) {
+        if (args.length !== 1) return;
+
+        // YYYY/MM/DD
+        const formatCheck = /\d{4}\/\d{2}\/\d{2}/;
+        if (!formatCheck.test(args[0])) return;
+
+        const data = args[0].split("/");
+        const date = `${data[0]} ${data[1]} ${data[2]}`;
+
+        this.sendPrintout(message.channel as Discord.TextChannel, this.schedule.get(date), date);
+    }
+
+    private postInfo() {
+        const esports = this.bot.guilds.get(this.settings.server)!.channels.find("name", "esports-spoilers");
+        if (!esports) {
+            console.error(`Esports: Unable to find channel #esports-spoilers`);
+            return;
+        }
+
+        const tomorrow = new Date();
+        tomorrow.setHours(tomorrow.getHours() + 12);
```

Comment:
Can you explain why only 12 hours are needed to get to tomorrow? Does this always work?

Reply:
Should be 24, but i didnt like that it output `in 2 days` from moment.js, so i reduced it 🤷‍♂️